tool-plugins/vscode/plugin/debugger/index.js

```diff
@@ -40,6 +40,7 @@ class BallerinaDebugSession extends LoggingDebugSession {
         this.debugManager.on('debug-hit', debugArgs => {
             const threadId = debugArgs.threadId || debugArgs.workerId;
             this.debugArgs[threadId] = debugArgs;
+            this.currentThreadId = threadId;
```

Comment:
On a multiworker program this will have problems. Because now the currentThreadId will be the worker id where the last breakpoint hit.